microcosm_flask/fields/order_by_field.py

```diff
@@ -0,0 +1,62 @@
+from enum import Enum
+
+from marshmallow import fields
+
+from microcosm_flask.fields import EnumField
+
+
+class OrderDirection(Enum):
+    ASC = "+"
+    DESC = "-"
+
+
+class OrderBy:
+
+    def __init__(self, order_field: Enum, order_dir: OrderDirection):
+        self.order_field = order_field
+        self.order_dir = order_dir
+
+    def __str__(self):
+        # This is used when serializing the object to a string for the URL. It doesn't actually use the _serialize
+        # method below.
+        return self.order_dir.value + self.order_field.name
+
+
+class OrderByField(fields.Str):
```

Comment:
One of my original versions had this extend the enum class, but then that gets into an issue when we generate the swagger defts as it only sets the valid enums as those passed in instead of something like "+ENUM". This isn't an issue server side, but breaks client side when using Bravado.